src/serial.rs

```diff
@@ -675,24 +906,19 @@ macro_rules! hal {
                     mut buffer: B,
                 ) -> CircBuffer<B, $rx_chan>
                 where
-                    B: StableDeref<Target = [H; 2]> + DerefMut,
+                    B: StableDeref<Target = [H; 2]> + DerefMut + 'static,
```

Comment:
Maybe change the `StableDeref` to just `Deref` ?

Reply:
Yeah I don't think `StableDeref` is required anymore; I can't actually remeber why I pulled it in to behonest. This change can probably wait though, I wouldn't want to block this PR over it.

Reply:
`StableDeref` kinda helps, but it isn't enough to get rid of the `mem::forget` problem. The way it is now (without the `'static`) is unsound. I guess it's a breaking change to put `'static`, but we have to do it.
We probably don't need `StableDeref` now that we got `'static`, so we might as well change it to `Deref`.